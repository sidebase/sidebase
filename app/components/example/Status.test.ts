import { describe, expect, test } from 'vitest'
import Status from './Status.vue'
import { render } from '~/tests/testingLibraryVue'
import type { StatusCheckFail, StatusCheckSuccess } from '~/types'

const fakeStatusCheckSuccess: StatusCheckSuccess = {
  error: null,
  result: {
    status: 'healthy',
    startupTime: new Date(2000, 1, 1, 13),
    time: new Date(2000, 1, 1, 14),
    nuxtAppVersion: 'v0.0.1',
  },
}

describe('Status', () => {
  test('mounts and is still the same for fail variant', () => {
    const statusCheck: StatusCheckFail = {
      error: true,
      result: null,
    }
    const { html } = render(Status, { props: { statusCheck } })
    expect(html()).toMatchSnapshot()
  })

  test('mounts and is still the same for success variant', () => {
    const { html } = render(Status, { props: { statusCheck: fakeStatusCheckSuccess } })
    expect(html()).toMatchSnapshot()
  })

  test('displays all status values', async () => {
    const { getByText } = render(Status, { props: { statusCheck: fakeStatusCheckSuccess } })

    // Ensure all expected text is there:
    getByText('Live backend service status is:')
    getByText('Status')
    getByText('healthy')
    getByText(`${fakeStatusCheckSuccess.result.time}`)
    getByText(`${fakeStatusCheckSuccess.result.startupTime}`)
    getByText(`${fakeStatusCheckSuccess.result.nuxtAppVersion}`)
  })
})
