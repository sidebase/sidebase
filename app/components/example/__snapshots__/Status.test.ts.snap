// Vitest Snapshot v1

exports[`Status > mounts and is still the same for fail variant 1`] = `
"<div>
  <p class=\\"font-bold\\"> An error occured. Health status could not be requested. </p>
</div>"
`;

exports[`Status > mounts and is still the same for success variant 1`] = `
"<div>
  <div>
    <div> Live backend service status is: </div>
    <table class=\\"table-fixed w-full\\">
      <tbody>
        <tr>
          <td class=\\"w-1/3\\"> Status </td>
          <td>healthy</td>
        </tr>
        <tr>
          <td>Server time</td>
          <td>Tue Feb 01 2000 14:00:00 GMT+0000 (Coordinated Universal Time)</td>
        </tr>
        <tr>
          <td>Startup Time</td>
          <td>Tue Feb 01 2000 13:00:00 GMT+0000 (Coordinated Universal Time)</td>
        </tr>
        <tr>
          <td>App version</td>
          <td>v0.0.1</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>"
`;
